gstd/src/async_runtime/locks.rs

```diff
@@ -0,0 +1,66 @@
+//! Wait duration registry
+use crate::{async_runtime, exec, prelude::BTreeMap, MessageId};
+
+/// Wait locks.
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) enum Lock {
+    For(u32),
+    NoMore(u32),
+}
+
+impl Lock {
+    /// Call wait functions by the lock type.
+    pub fn wait(&self) {
+        match self {
+            Lock::For(d) => exec::wait_for(*d),
+            Lock::NoMore(d) => exec::wait_no_more(*d),
+        }
+    }
+}
+
+impl Default for Lock {
+    fn default() -> Self {
+        unsafe { Lock::NoMore(crate::config::DEFAULT_WAIT_NO_MORE_DURATION) }
+    }
+}
+
+/// Wait trait for async sending messages.
+pub trait Wait: Sized {
```

Comment:
There are several problems with this trait:
1. We should always import this: I don't find this beauty for our contacts scheme
2. Doc comments of methods are messed up: `no_more` is about as much as possible, `till` about concrete waiting duration
3. `till(block_number)` naming for me means that we will wait until `block_number` comes: I'd suggest rename it into `exactly(..)` if you want to keep single word (I usually prefer it too)